Sources/Foundation/FileManager+POSIX.swift

```diff
@@ -346,8 +346,16 @@ extension FileManager {
                         try createDirectory(atPath: parent, withIntermediateDirectories: true, attributes: attributes)
                     }
                     if mkdir(pathFsRep, S_IRWXU | S_IRWXG | S_IRWXO) != 0 {
-                        throw _NSErrorWithErrno(errno, reading: false, path: path)
-                    } else if let attr = attributes {
+                        let posixError = errno
+                        if posixError == EEXIST && fileExists(atPath: path, isDirectory: &isDir) && isDir.boolValue {
```

Comment:
I think the test for `EEXIST` is only needed on the last path component to check if is a file.

If an earlier `mkdir()` fails because the existing path is a  file there should be an `ENOTDIR` error returned, which should also handle the edge case of `path` being a directory when `fileExists` is called but being replaced by a file before the next `mkdir()`.

Would need an extra test with a pre-existing path with includes a file as the last component to check this.

Reply:
`EEXIST` is returned by `mkdir` if the last path component already exists, regardless of whether it is a file or a directory. `ENOTDIR` is returned if any of the path components aside from the last one is not a directory.

This is supported by the [documentation](https://pubs.opengroup.org/onlinepubs/9699919799/functions/mkdir.html) and by the sample code in [SR-12272](https://bugs.swift.org/browse/SR-12272).